arangod/MMFiles/MMFilesWalRecoverState.cpp

```diff
@@ -381,6 +382,14 @@ bool MMFilesWalRecoverState::InitialScanMarker(MMFilesMarker const* marker,
           state->maxRevisionId = revisionId;
         }
       }
+
+      if (marker->getSize() == MMFilesDatafileHelper::VPackOffset(TRI_DF_MARKER_VPACK_DOCUMENT) + payloadSlice.byteSize() + sizeof(LocalDocumentId::BaseType)) {
+        // we do have a LocalDocumentId stored at the end of the marker
+        uint8_t const* ptr = payloadSlice.begin() + payloadSlice.byteSize();
+        LocalDocumentId localDocumentId{encoding::readNumber<LocalDocumentId::BaseType>(ptr, sizeof(LocalDocumentId::BaseType))};
+        LocalDocumentId::track(localDocumentId);
```

Comment:
would it make sense to only keep a local copy of the maximum LocalDocumentId value seen so far, and calling `LocalDocumentId::track(...)` only once at the end of the scan procedure? That would save many many calls to `track` for bigger collections.

Reply:
Yes, probably. That assumes that the generation of IDs is monotonically increasing and we only need to track the largest. This, of course, is true for our current method of ID generation, and I'm guessing it will continue to be so in the future, so it should be fine.